swap/int256/int256.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2020 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package int256
+
+import (
+	"fmt"
+	"io"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// Int256 represents an signed integer of 256 bits
+type Int256 struct {
+	value big.Int
+}
+
+var minInt256 = new(big.Int).Mul(big.NewInt(-1), new(big.Int).Exp(big.NewInt(2), big.NewInt(255), nil)) // -(2^255)
+var maxInt256 = new(big.Int).Sub(new(big.Int).Exp(big.NewInt(2), big.NewInt(255), nil), big.NewInt(1))  // 2^255 - 1
+
+// NewInt256 creates a Int256 struct with an initial underlying value of the given param
+// returns an error when the result falls outside of the signed 256-bit integer range
+func NewInt256(value big.Int) (*Int256, error) {
+	u := new(Int256)
+	return u.set(value)
+}
+
+// Int256From creates a Int256 struct based on the given int64 param
+// any int64 is valid as a Int256
+func Int256From(base int64) *Int256 {
+	u := new(Int256)
+	u.value = *new(big.Int).SetInt64(base)
```

Comment:
so, every time you call a line which has this syntax you are effectively telling the runtime to:
1. allocate an `Int256` which encapsulates a `big.Int`
2. allocate another `big.Int`, which is a struct of an arbitrary size of `uint8` words and a bool, plus some boxing.
3. then, you call `SetInt64`, which effectively does the actual allocation on the _stack_ for the word slice in `big.Int`. 
4. then, you dereference the pointer, resulting in another allocation on the stack
5. the value of the dereferenced pointer is then copied to the value of the `Int256` you allocated in line `45`.

So effectively you're creating 4 allocations in 3 lines of code, just in order to use this one value. this is causing unnecessary pressure on memory management.
@janos has suggested to call `u.value.SetInt64` directly. why not do that?

Reply:
you're absolutely right. i've changed the functions so that they call `SetInt64` from `u.value` directly

Reply:
thanks <3